### ISSUE TITLE
The new fine-grained staters idea is bad

### ISSUE THREAD

**Issue by hantsy** [togglz/togglz#1035]
I do not think the new staters added in the main branch is good, too many staters will confuse users.

For togglz, I only need on starter if only supports webmvc, it is enough,  all smaller features can be categorized into a standalone configuration classes.

Of course, I hope starting a webflux stater for the Spring WebFlux/Reactive stack.

* togglz-spring-boot-stater-webmvc
* togglz-spring-boot-stater-webflux

**Comment by bennetelli** [comment]
cc @cfranzen

**Comment by bennetelli** [comment]
you can still use the "old" one :) I will discuss it with some more users and think about it again. Thanks for opening this issue :)

**Comment by cfranzen** [comment]
@hantsy please see my comment on PR #888 explaining why I implemented these fine-grained starters: https://github.com/togglz/togglz/pull/888#issuecomment-1305245134

**Comment by hantsy** [comment]
@cfranzen  Spring Boot itself has a thymeleaf starter, why creating one in this project https://github.com/togglz/togglz/tree/master/spring-boot/starter-thymeleaf

Thymeleaf is related to console and will introduce transitive dependencies, make it as a standalone starter is reasonable.

* togglz-autoconfigure(optional module to includes all autoconfiguraitons for all smaller features)
* togglz-webmvc-starter (including base core, web, security)
* togglz-console-starter ( depends on togglz-webmvc-starter directly, or pick up different autoconfigurations)

**Comment by hantsy** [comment]
> @hantsy please see my comment on PR #888 explaining why I implemented these fine-grained starters: [#888 (comment)](https://github.com/togglz/togglz/pull/888#issuecomment-1305245134)
> 
> Furthermore, it would be simple to provide another spring boot starter that combines all the other starters. If you don't care about your dependencies, you could use this one.

In our project, we are using WebFlux/Kotlin Coroutines, besides the core autoconfiguration, others are no use at all. Currently we used the in-memory implementation of the state repository, but it seems there is no WebFlux support and Kotlin Coroutines support.

**Comment by hantsy** [comment]
> you can still use the "old" one :) I will discuss it with some more users and think about it again. Thanks for opening this issue :)

Yeah, we used it, but excluded servlet, console, web, etc.  Only required the core autoconfigure. We are using Spring WebFlux/Kotlin Coroutines.

**Comment by cfranzen** [comment]
> > you can still use the "old" one :) I will discuss it with some more users and think about it again. Thanks for opening this issue :)
> 
> Yeah, we used it, but excluded servlet, console, web, security, etc. Only required the core autoconfigure. We are using Spring WebFlux/Kotlin Coroutines.

I am a bit confused now that you complain about the new Spring Boot Starters, since the use case you explain here is exactly what the tooglz-spring-boot-starter-core is designed for. I have the same use case. We are doing Kotlin and Webflux in a Microservice landscape and do not want to expose any Togglz endpoint. Mainly for security reasons, but your use case is equally valid. With the new Spring Boot starters you just can have a dependency on tooglz-spring-boot-starter-core without the need to maintain any dependency excludes.

**Comment by cfranzen** [comment]
> @cfranzen Spring Boot itself has a thymeleaf starter, why creating one in this project https://github.com/togglz/togglz/tree/master/spring-boot/starter-thymeleaf
> 
> Thymeleaf is related to console and will introduce transitive dependencies, make it as a standalone starter is reasonable.
> 
> * togglz-autoconfigure(optional module to includes all autoconfiguraitons for all smaller features)
> * togglz-webmvc-starter (including base core, web, security)
> * togglz-console-starter ( depends on togglz-webmvc-starter directly, or pick up different autoconfigurations)

It is quite simple. I don't want my library for feature toggles to add a transitive dependency to thymeleaf to my project. The rational behind Spring Boot starters is to simplify dependency management, but with your suggestion above, it would become even harder. 

Example: I have a simple web application and want to use your togglz-webmvc-starter. That would automatically add the dependency to Spring Security which would trigger all the Spring Security auto-configuration stuff. Just by adding the dependency to togglz-webmvc-starter it would be required to think about HTTP authentication stuff and CSRF token protection. In my opinion just a bad solution.
